### PR TITLE
Add fully Hypixel-Rule compliant glowing mush highlighter

### PR DESCRIPTION
**Pull request by romangraef**

### CODE REVIEW

**Review by hannibal00212 — CHANGES_REQUESTED** [review]
test

**Review by hannibal00212 — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+
+@Cancelable
+public class SpawnParticleEvent extends NEUEvent {
+
+	// I LOVE code duplication!
+	// I LOVE repeating field names five times to declare a data class!
+	// I LOVE that yummy high fructose corn syrup!
+	// I LOVE the United Nations!
```
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java:31]: ???

Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java:31]: Do you not appreciate the United Nations, just like I do, @IRONM00N?

File: src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+package io.github.moulberry.notenoughupdates.events;
+
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+
+@Cancelable
+public class SpawnParticleEvent extends NEUEvent {
+
+	// I LOVE code duplication!
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java:28]: Please remove these comments.

Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java:28]: Please remove these comments.

**Review by hannibal002 — CHANGES_REQUESTED** [review]

**Review by hannibal002 — CHANGES_REQUESTED** [review]
We are no cheaters here.
grafik

I don't think non-tech-savvy users will understand that category name
grafik

Maybe only render the hitbox of the mushroom and not the full 1x1x1 block.
And, please change the default colour to something more in the eye pointy (green maybe?)

**Comment by romangraef** [comment]
it only renders through walls if you could previously see them (keeps a list)

**Comment by hannibal002** [comment]
@romangraef merge needed

File: src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java (hunk 2 above)
Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/events/SpawnParticleEvent.java:28]: No

**Comment by walkerselby** [comment]
> it only renders through walls if you could previously see them (keeps a list)

Can we add a message in the config saying something to the extent of "it only renders through walls if you could previously see them (keeps a list)" so we don't get a bunch of messages like we are with Alpha about ESP? 

And maybe a toggle for the through wall if you feel like it?

Obviously is not a cheat, but y'know, SkyBlock players and idiocy and presumptions

**Comment by hannibal002** [comment]
> Obviously is not a cheat, but y'know, SkyBlock players and idiocy and presumptions

👍

**Comment by romangraef** [comment]
I dont see the point in adding a warning; I doubt most people would even read such a message.

**Comment by hannibal002** [comment]
> I dont see the point in adding a warning; I doubt most people would even read such a message.

People are lazy.

**Review by hannibal002 — APPROVED** [review]
No need to change stuff since people are lazy and Hypixel is even lazier in enforcing or even checking stuff.

**Comment by jani270** [comment]
> No need to change stuff since people are lazy and Hypixel is even lazier in enforcing or even checking stuff.

I dont think thats a good idea.

**Comment by NopoTheGamer** [comment]
Yeah idk, I would prefer if it didn't really show through walls

**Review by hannibal002 — CHANGES_REQUESTED** [review]
I would prefer more feedback from neu to my change suggestions from 10 days ago:

**Comment by GodOfProDev** [comment]
> Yeah idk, I would prefer if it didn't really show through walls

I am pretty sure it only shows through walls if you previously saw the mushroom

**Comment by jani270** [comment]
> > Yeah idk, I would prefer if it didn't really show through walls
> 
> I am pretty sure it only shows through walls if you previously saw the mushroom

Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.

**Comment by GodOfProDev** [comment]
> > > Yeah idk, I would prefer if it didn't really show through walls
> > 
> > 
> > I am pretty sure it only shows through walls if you previously saw the mushroom
> 
> Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.

a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough

**Comment by jani270** [comment]
> > > > Yeah idk, I would prefer if it didn't really show through walls
> > > 
> > > 
> > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > 
> > 
> > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> 
> a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough

Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.

**Comment by GodOfProDev** [comment]
> > > > > Yeah idk, I would prefer if it didn't really show through walls
> > > > 
> > > > 
> > > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > > 
> > > 
> > > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> > 
> > 
> > a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough
> 
> Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.

so you say just remove the feature entirely?

**Comment by IRONM00N** [comment]
> > > > > Yeah idk, I would prefer if it didn't really show through walls
> > > > 
> > > > 
> > > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > > 
> > > 
> > > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> > 
> > 
> > a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough
> 
> Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.

You don't know that, you will not receive support from people who did actually bother reading the disclaimer, I don't think not proving a disclaimer would be beneficial.

**Comment by jani270** [comment]
> so you say just remove the feature entirely?

Im saying that see through walls shouldnt be a part of this feature.

**Comment by jani270** [comment]
> > > > > > Yeah idk, I would prefer if it didn't really show through walls
> > > > > 
> > > > > 
> > > > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > > > 
> > > > 
> > > > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> > > 
> > > 
> > > a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough
> > 
> > 
> > Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.
> 
> You don't know that, you will not receive support from people who did actually bother reading the disclaimer, I don't think not proving a disclaimer would be beneficial.

What? We will see support tickets from it because people dont read disclaimers? Thats the entire point. Its not that big of deal. We will at some point have people crying in forums and in our support. And yes i know that, because ive seen that in other cases (not neu releated)

**Comment by IRONM00N** [comment]
> > > > > > > Yeah idk, I would prefer if it didn't really show through walls
> > > > > > 
> > > > > > 
> > > > > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > > > > 
> > > > > 
> > > > > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> > > > 
> > > > 
> > > > a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough
> > > 
> > > 
> > > Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.
> > 
> > 
> > You don't know that, you will not receive support from people who did actually bother reading the disclaimer, I don't think not proving a disclaimer would be beneficial.
> 
> What? We will see support tickets from it because people dont read disclaimers? Thats the entire point. Its not that big of deal. We will at some point have people crying in forums and in our support. And yes i know that, because ive seen that in other cases (not neu releated)

Then just add another option that toggles this behavior that is off by default. It could be called "cache previously viewed mushrooms" or something.

**Comment by GodOfProDev** [comment]
> > > > > > > > Yeah idk, I would prefer if it didn't really show through walls
> > > > > > > 
> > > > > > > 
> > > > > > > I am pretty sure it only shows through walls if you previously saw the mushroom
> > > > > > 
> > > > > > 
> > > > > > Yes we know that. But Hypixel Players are hypixel players, most of them cant tell the difference and we all know the forums.
> > > > > 
> > > > > 
> > > > > a simple fix would be just adding a disclaimer in the description like ( only shows through walls if you have seen it before ) not sure if its enough
> > > > 
> > > > 
> > > > Wrong, noone reads disclaimer, see Skill Overlays. There is a disclaimer that you need a enchant and people still ask why its not working.
> > > 
> > > 
> > > You don't know that, you will not receive support from people who did actually bother reading the disclaimer, I don't think not proving a disclaimer would be beneficial.
> > 
> > 
> > What? We will see support tickets from it because people dont read disclaimers? Thats the entire point. Its not that big of deal. We will at some point have people crying in forums and in our support. And yes i know that, because ive seen that in other cases (not neu releated)
> 
> Then just add another option that toggles this behavior that is off by default. It could be called "cache previously viewed mushrooms" or something.

i think adding a toggle would be a better option than entirely removing it

**Comment by romangraef** [comment]
image
hell thread btw

**Comment by jani270** [comment]
> image hell thread btw

I dont see an issue here

**Comment by IRONM00N** [comment]
> > image hell thread btw
> 
> I dont see an issue here

Me neither

**Comment by romangraef** [comment]
im just gonna make it only visible with direct sightline, since [this](https://git.nea.moe/nea/neuhax/src/branch/master/src/main/java/moe/nea/sky/mixin/config/MixinHaxConfigWorld.java) exists

**Review by hannibal002 — DISMISSED** [review]

**Review by NopoTheGamer — DISMISSED** [review]

**Review by hannibal002 — DISMISSED** [review]

**Review by Lulonaut — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**